infra/modules/monitoring/main.tf

```diff
@@ -0,0 +1,126 @@
+  ok_actions                = [aws_sns_topic.this.arn]
+  insufficient_data_actions = [aws_sns_topic.this.arn]
+
+  dimensions = {
+    ClusterName = var.cluster_name
+    ServiceName = var.service_name
+  }
+}
+
+resource "aws_cloudwatch_metric_alarm" "memory_utilization_alarm" {
+  count                     = var.high_memory_util_threshold != null ? 1 : 0
+  alarm_name                = "${var.cluster_name}-memory-utilization"
+  comparison_operator       = "GreaterThanThreshold"
+  evaluation_periods        = 3
+  metric_name               = "MemoryUtilization"
+  namespace                 = "AWS/ECS"
+  period                    = 300
+  statistic                 = "Average"
+  threshold                 = var.high_memory_util_threshold
+  alarm_description         = "Alarm triggered when memory utilization is above 70% for 15 minutes."
+  alarm_actions             = [aws_sns_topic.this.arn]
+  ok_actions                = [aws_sns_topic.this.arn]
+  insufficient_data_actions = [aws_sns_topic.this.arn]
+  dimensions = {
+    ClusterName = var.cluster_name
+    ServiceName = var.service_name
+  }
+}
+
+resource "aws_cloudwatch_metric_alarm" "task_health_alarm" {
+  count                     = var.task_health_percentage_threshold != null ? 1 : 0
+  alarm_name                = "${var.cluster_name}-task-health"
+  comparison_operator       = "LessThanThreshold"
+  evaluation_periods        = 1
+  metric_name               = "TaskHealthPercentage"
+  namespace                 = "AWS/ECS"
+  period                    = 60
+  statistic                 = "Minimum"
+  threshold                 = var.task_health_percentage_threshold
+  alarm_description         = "Alarm triggered when any ECS task becomes unhealthy."
+  alarm_actions             = [aws_sns_topic.this.arn]
+  ok_actions                = [aws_sns_topic.this.arn]
+  insufficient_data_actions = [aws_sns_topic.this.arn]
+  dimensions = {
+    ClusterName = var.cluster_name
+  }
+}
+
+resource "aws_cloudwatch_metric_alarm" "task_placement_errors_alarm" {
+  count                     = var.task_placement_error_threshold != null ? 1 : 0
+  alarm_name                = "${var.cluster_name}-task-placement-errors"
+  comparison_operator       = "GreaterThanThreshold"
+  evaluation_periods        = 3
+  metric_name               = "TaskPlacementError"
+  namespace                 = "AWS/ECS"
+  period                    = 300
+  statistic                 = "Sum"
+  threshold                 = var.task_placement_error_threshold
+  alarm_description         = "Alarm triggered when ECS tasks encounter placement errors."
+  alarm_actions             = [aws_sns_topic.this.arn]
+  ok_actions                = [aws_sns_topic.this.arn]
+  insufficient_data_actions = [aws_sns_topic.this.arn]
+  dimensions = {
+    ClusterName = var.cluster_name
+  }
+}
+
+resource "aws_cloudwatch_metric_alarm" "service_availability_alarm" {
+  count                     = var.service_availability_threshold != null ? 1 : 0
```

Comment:
this alarm also seems like something that's worth just having by default rather than being able to turn on/off. i think it's worth notifying project teams if any of the service is unavailable.

question: is this redundant with task availability or different? how is service availability measured?

Reply:
as we agreed those alerts were removed.